File: src/schemas/trip.py
from __future__ import annotations

import datetime as dt
from decimal import Decimal

from pydantic import Field

from src.schemas.common import ORMModel


class TripBase(ORMModel):
    company_id: int
    date: dt.date
    freight: str = Field(max_length=25)
    origin: str = Field(max_length=255)
    destination: str = Field(max_length=255)
    destination_company_name: str | None = Field(default=None, max_length=255)
    trip_category: str = Field(default="domestic", max_length=20)
    amount: Decimal
    toll_gate: Decimal = Decimal("0.00")
    driver: str = Field(max_length=30)
    driver_id: int | None = None
    external_driver_name: str | None = Field(default=None, max_length=120)
    external_driver_mobile: str | None = Field(default=None, max_length=25)


class TripCreate(TripBase):
    pass


class TripUpdate(ORMModel):
    date: dt.date | None = None
    freight: str | None = Field(default=None, max_length=25)
    origin: str | None = Field(default=None, max_length=255)
    destination: str | None = Field(default=None, max_length=255)
    destination_company_name: str | None = Field(default=None, max_length=255)
    trip_category: str | None = Field(default=None, max_length=20)
    amount: Decimal | None = None
    toll_gate: Decimal | None = None
    driver: str | None = Field(default=None, max_length=30)
    driver_id: int | None = None
    external_driver_name: str | None = Field(default=None, max_length=120)
    external_driver_mobile: str | None = Field(default=None, max_length=25)
    paid: bool | None = None


class TripRead(TripBase):
    id: int
    vat: Decimal
    total_amount: Decimal
    paid: bool
    invoice_id: int | None

File: frontend/src/types/trip.ts
export interface TripApi {
  id: number;
  company_id: number;
  date: string;
  freight: string;
  origin: string;
  destination: string;
  destination_company_name?: string | null;
  trip_category?: "domestic" | "international";
  amount: string;
  toll_gate: string;
  driver: string;
  driver_id?: number | null;
  external_driver_name?: string | null;
  external_driver_mobile?: string | null;
  vat: string;
  total_amount: string;
  paid: boolean;
  invoice_id: number | null;
  created_at?: string;
  company_name?: string;
}

export interface TripCreateInput {
  companyId: number;
  date: string;
  freight: string;
  origin: string;
  destination: string;
  destinationCompanyName?: string;
  tripCategory: "domestic" | "international";
  amount: number;
  tollGate: number;
  driver: string;
  driverId?: number;
  externalDriverName?: string;
  externalDriverMobile?: string;
}
